components/CMakeLists.txt

```diff
@@ -104,6 +104,16 @@ elseif(USE_HIP)
   enable_language(HIP)
 endif()
 
+# Any changes to SourceMods will require us to reconfigure
+file(GLOB COMPONENT_SOURCE_MOD_DIRS "${CASEROOT}/SourceMods/src.*")
+foreach(COMPONENT_SOURCE_MOD_DIR IN LISTS COMPONENT_SOURCE_MOD_DIRS)
+  set_property(
+    DIRECTORY
+    APPEND
+    PROPERTY CMAKE_CONFIGURE_DEPENDS
+    ${COMPONENT_SOURCE_MOD_DIR})
+endforeach()
+
```

Comment:
Thank you @jgfouca . This change does bring back the SourceMods mechanism for model build.
A question on the comment: "`Any changes to SourceMods will require us to reconfigure`"? Can you give some explanation?  Is it the new way of making SourceMods work? My understanding has been that no reconfigure is involved when making changes via SourceMods. The "`reconfigure`" you refer to probably has a different meaning.

Reply:
@wlin7 , yes, I will try to explain more. With this change the CONFIGURE phase of cmake now depends on the SourceMod dirs for the various components. Any file change within those directories (adding or removing a file) will change the mod timestamp for the directory. This will allow CMake to "see" those kinds of changes and know it needs to re-run itself.